pandas/_libs/tslibs/offsets.pyx

```diff
@@ -187,7 +187,7 @@ def apply_wraps(func):
             result = result.normalize()
 
         # nanosecond may be deleted depending on offset process
```

Comment:
do we have any test cases with both nano !=0 and hasattr(self, "nanoseconds")?

if you can make this comment (L189) more specific that'd be a nice bonus

Reply:
Yes, we have. It's in `pandas/tests/tseries/offsets/test_offsets.py` L816, let me add an comment and mention you in that line.

Reply:
Also added the explanation for L189.